app/src/App.tsx

```diff
@@ -25,6 +26,8 @@ import Copyable from './components/Copyable';
 const DRAWER_WIDTH = '40vw';
 
 function App() {
+  // Current theme state
+  const [theme,setTheme]=useState<Theme>('light');
```

Comment:
Rather than setting the default to light, we should just call `useContext` here to get the theme context:

```
  const theme = useContext(ThemeContext);
```

Reply:
I'm afraid with that I won't be able to get & set a theme.

What do you think if I remove in `ActionToolbar.tsx`

```
useEffect(() => {
    if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
      setTheme('dark');
    } else {
      setTheme('light');
    }
  }, []);
```
And make a theme check inside `App.tsx`

```
const [theme, setTheme] = useState<Theme>(window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light');
```